### DISCUSSION TITLE
Subtracting 1 to achieve matching array length

### DISCUSSION THREAD

**Discussion by monolithMktg** [cables-gl/cables_docs#358] [Answers & Questions] [answered]
I was watching [this ](https://www.youtube.com/watch?v=mQN8VtVOltQ&t=652s)tutorial on **Texture2ColorArray** Op, and towards 8:20 in the video, **_Andro_** subtracts 1 from number of rows and columns on the rectangle to match the array length of the **Texture2ColorArray** length.

Why was this required? Why was one row and column acting as extra?

**Comment by steam0r** [comment, marked as answer]
0-based vs. 1-based counting

**Reply by monolithMktg** [reply]
Ohh makes sense. Thank you :)
